booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,654 @@
+        if (!partyFunctions.contains("BA")) {
+          issues.add("The 'BA' party must be provided if 'OS' or 'DDR' are absent ");
+        }
+      }
+        return issues;
+    });
+
+  private static final List<JsonContentCheck> STATIC_BOOKING_CHECKS = Arrays.asList(
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/cargoMovementTypeAtOrigin"), BookingDataSets.CARGO_MOVEMENT_TYPE),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/cargoMovementTypeAtDestination"), BookingDataSets.CARGO_MOVEMENT_TYPE),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/communicationChannelCode"), BookingDataSets.COMMUNICATION_CHANNEL_CODES),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/declaredValueCurrency"), BookingDataSets.ISO_4217_CURRENCY_CODES),
+    JsonAttribute.mustBeDatasetKeywordIfPresent(JsonPointer.compile("/incoTerms"), BookingDataSets.INCO_TERMS_VALUES),
+    JsonAttribute.atMostOneOf(
+      JsonPointer.compile("/shippedOnBoardDate"),
+      JsonPointer.compile("/receivedForShipmentDate")
+    ),
+    CHECK_EXPECTED_DEPARTURE_DATE,
+    CHECK_EXPECTED_ARRIVAL_POD,
+    NOR_PLUS_ISO_CODE_IMPLIES_ACTIVE_REEFER,
+    IS_SHIPPER_OWNED_CONTAINER,
+    REFERENCE_TYPE_VALIDATION,
+    ISO_EQUIPMENT_CODE_VALIDATION,
+    IS_EXPORT_DECLARATION_REFERENCE_PRESENT,
+    IS_IMPORT_DECLARATION_REFERENCE_PRESENT,
+    OUTER_PACKAGING_CODE_IS_VALID,
+    TLR_CC_T_COMBINATION_VALIDATIONS,
+    DOCUMENT_PARTY_FUNCTIONS_MUST_BE_UNIQUE,
+    UNIVERSAL_SERVICE_REFERENCE,
+    VALIDATE_SHIPMENT_CUTOFF_TIME_CODE,
+    VALIDATE_ALLOWED_SHIPMENT_CUTOFF_CODE,
+    VALIDATE_DOCUMENT_PARTIES,
+    JsonAttribute.atLeastOneOf(
+      JsonPointer.compile("/expectedDepartureDate"),
+      JsonPointer.compile("/expectedArrivalAtPlaceOfDeliveryStartDate"),
+      JsonPointer.compile("/expectedArrivalAtPlaceOfDeliveryEndDate"),
+      JsonPointer.compile("/carrierExportVoyageNumber")
+    ),
+    JsonAttribute.allIndividualMatchesMustBeValid(
+      "DangerousGoods implies packagingCode or imoPackagingCode",
+      mav -> mav.submitAllMatching("requestedEquipments.*.commodities.*.outerPackaging"),
+      (nodeToValidate, contextPath) -> {
+        var dg = nodeToValidate.path("dangerousGoods");
+        if (!dg.isArray() || dg.isEmpty()) {
+          return Set.of();
+        }
+        if (nodeToValidate.path("packageCode").isMissingNode() && nodeToValidate.path("imoPackagingCode").isMissingNode()) {
+          return Set.of("The '%s' object did not have a 'packageCode' nor an 'imoPackagingCode', which is required due to dangerousGoods"
+            .formatted(contextPath));
+        }
+        if (nodeToValidate.path("numberOfPackages").isMissingNode() && nodeToValidate.path("description").isMissingNode()) {
+          return Set.of("The '%s' object did not have a 'numberOfPackages' nor an 'description', which is required due to dangerousGoods"
```

Comment:
This issue does not match the title of the check.  Consider splitting the check into two.

Reply:
(also, if the split is not done, then both issues should be emittable at the same time as they are not dependent on each other)

Reply:
It looks like you attempted to split out of the change, but I still see the number of packages vs. description check embedded in the packagingCode/imoPackagingCode check

Reply:
missed to delete it.. thanks for pointing